python/paddle/fluid/transpiler/details/program_utils.py

```diff
@@ -21,13 +21,17 @@
 
 
 def delete_ops(block, ops):
-    try:
-        start = list(block.ops).index(ops[0])
-        end = list(block.ops).index(ops[-1])
-        [block._remove_op(start) for _ in six.moves.range(end - start + 1)]
-    except Exception as e:
-        raise e
-    block.program._sync_with_cpp()
+    for op in ops:
+        try:
+            idx = list(block.ops).index(op)
+            block._remove_op(idx)
+            # del block.ops[idx]
+            # end = list(block.ops).index(ops[-1])
+            # [block._remove_op(start) for _ in six.moves.range(end - start + 1)]
+        except Exception as e:
+            print(e)
+        # raise e
+        # block.program._sync_with_cpp()
```

Comment:
Do we need call the `sync_with_cpp()` function? And please delete the unused code.